src/main/java/com/ibm/crypto/plus/provider/RSASignature.java

```diff
@@ -84,6 +86,22 @@ protected void engineInitSign(PrivateKey privateKey) throws InvalidKeyException
             throw new InvalidKeyException("Key is not an RSAPrivateKey");
         }
 
+        try {
+            if (provider.isFIPS()) {
+                RSAKeyFactory.checkKeyLengths(((java.security.interfaces.RSAPrivateKey) privateKey).getModulus().bitLength(),
+                        RSAKeyGenParameterSpec.F4,
```

Comment:
Could we limit the size of the modulus used for verification also? I believe that this should be 1024, 2048, 3072, and 4096 for FIPS. We may need an additional test to validate these sizes for the RSA FIPS verification process.

Reply:
limited